src/Server/TextDocument.php

```diff
@@ -79,7 +57,7 @@ public function documentSymbol(TextDocumentIdentifier $textDocument): array
      */
     public function didOpen(TextDocumentItem $textDocument)
     {
-        $this->updateAst($textDocument->uri, $textDocument->text);
+        $this->project->getDocument($textDocument->uri)->updateAst($textDocument->text);
```

Comment:
In case this proves to be a long blocking call, we could also think about queuing that to the end of the event loop and returning a promise.